cloudservice.go

```diff
@@ -8,11 +8,18 @@ import (
 	"github.com/juju/schema"
 )
 
+// CloudService represents the state of a CAAS service.
+type CloudService interface {
+	ProviderId() string
+	Addresses() []Address
+	SetAddresses(addresses []AddressArgs)
+}
+
 type cloudService struct {
 	Version int `yaml:"version"`
 
-	ProviderId_ string   `yaml:"provider-id,omitempty"`
-	Addresses_  []string `yaml:"addresses,omitempty"`
+	ProviderId_ string     `yaml:"provider-id,omitempty"`
+	Addresses_  []*address `yaml:"addresses,omitempty"`
```

Comment:
Similarly it seems odd to have a slice of pointers to addresses.

Reply:
yeah, good point so here they should not be pointers, i'll fix

Reply:
actually, sorry, it needs to be a pointer, as it is for machine addresses etc elsewhere
the pointer is what implements the Address interface